packages/flutter/lib/src/material/scrollbar.dart

```diff
@@ -43,182 +49,105 @@ class Scrollbar extends StatefulWidget {
+  Animation<double> _fadeoutOpacityAnimation;
+  Timer _fadeoutTimer;
 
   @override
-  void dispose() {
-    _painter.dispose();
-    super.dispose();
-  }
-
-  @override
-  Widget build(BuildContext context) {
-    return new NotificationListener<ScrollNotification>(
-      onNotification: _handleScrollNotification,
-      // TODO(ianh): Maybe we should try to collapse out these repaint
-      // boundaries when the scroll bars are invisible.
-      child: new RepaintBoundary(
-        child: new CustomPaint(
-          foregroundPainter: _painter,
-          child: new RepaintBoundary(
-            child: widget.child,
-          ),
-        ),
-      ),
+  void initState() {
+    super.initState();
+    _fadeoutAnimationController = new AnimationController(
+      vsync: this,
+      duration: _kScrollbarFadeDuration,
+    );
+    _fadeoutOpacityAnimation = new CurvedAnimation(
+      parent: _fadeoutAnimationController,
+      curve: Curves.fastOutSlowIn
     );
-  }
-}
-
-class _ScrollbarPainter extends ChangeNotifier implements CustomPainter {
-  _ScrollbarPainter(TickerProvider vsync)
-    : assert(vsync != null) {
-    _fadeController = new AnimationController(duration: _kThumbFadeDuration, vsync: vsync);
-    _opacity = new CurvedAnimation(parent: _fadeController, curve: Curves.fastOutSlowIn)
-      ..addListener(notifyListeners);
-  }
-
-  // animation of the main axis direction
-  AnimationController _fadeController;
-  Animation<double> _opacity;
-
-  // fade-out timer
-  Timer _fadeOut;
-
-  Color get color => _color;
-  Color _color;
-  set color(Color value) {
-    assert(value != null);
-    if (_color == value)
-      return;
-    _color = value;
-    notifyListeners();
-  }
-
-  TextDirection get textDirection => _textDirection;
-  TextDirection _textDirection;
-  set textDirection(TextDirection value) {
-    assert(value != null);
-    if (_textDirection == value)
-      return;
-    _textDirection = value;
-    notifyListeners();
   }
 
   @override
-  void dispose() {
-    _fadeOut?.cancel();
-    _fadeController.dispose();
-    super.dispose();
-  }
+  void didChangeDependencies() {
+    super.didChangeDependencies();
 
-  ScrollMetrics _lastMetrics;
-  AxisDirection _lastAxisDirection;
-
-  static const double _kMinThumbExtent = 18.0;
-  static const double _kThumbGirth = 6.0;
-  static const Duration _kThumbFadeDuration = const Duration(milliseconds: 300);
-  static const Duration _kFadeOutTimeout = const Duration(milliseconds: 600);
-
-  void update(ScrollMetrics metrics, AxisDirection axisDirection) {
-    _lastMetrics = metrics;
-    _lastAxisDirection = axisDirection;
-    if (_fadeController.status == AnimationStatus.completed) {
-      notifyListeners();
-    } else if (_fadeController.status != AnimationStatus.forward) {
-      _fadeController.forward();
+    final ThemeData theme = Theme.of(context);
+    _currentPlatform = theme.platform;
+
+    if (_currentPlatform == TargetPlatform.iOS) {
```

Comment:
here and in various other places: where you currently us an `if` to check the platform, use a `switch` so that the analyzer tells us if we missed handling a new platform when we add one

Reply:
Done